src/controllers/tourController.js

```diff
@@ -36,15 +36,33 @@ exports.fetchSingleTour = (req, res) => {
 }
 
 // book a tour
-exports.store = async (req, res) => {
-	// TODO: get user id from url
-	// instantiating new booking
-	const newBooking = new Booking({...req.body});
-	// saving new booking
-	const bookedTour = await newBooking.save();
-	console.log(bookedTour);
-	// send confirmation email
-	await sendConfirmationEmail(bookedTour, req, res);
+exports.bookTour = async (req, res) => {
+	try {
+		// get user id from url
+		let loggedInUser = await User.findById(req.params.id);
+
+		// search for existing tour by id, if no tour exists, create one with logged in user info.
+		Tour.findById(req.body.tour, (err, tour) => {
+			if (err) {
+				return res.status(404).json({message: 'Tour not found'});
+			} else {
+				Booking.create({
```

Comment:
same here

Reply:
I tried using await here but cant use await nested inside another await: "The 'await' operator can only be used in an 'async' function"

Reply:
So the concept of using `async` and `await` in this scenario is that you will not need to use callbacks again. For example, instead of 

```
User.findById(req.user.id, (err, user) => {
if(err){
...
} else{
//... do something with "user"
}
}
```

You can use:
`let user = await User.findById(req.user.id)`;
you will only have "user" (which you would have passed in the callback before) if User.findById happens without an error (promise rejection). You may ask, what if there's an error? That's where `catch(err){...}` comes in. Any error which you would have caught by callbacks get caught in that block.
So I'd rewrite your code as:
```try{
let loggedInUser = await User.findById(req.user.id);
if(!loggedInUser){
    //...do something
    };
let tour = await Tour.findById(req.body.tour);
if(!tour){
    //...do something
} else{
    let newBooking = await Booking.create({...})
   loggedInUser.bookedTours.push(newBooking);
   await loggedInUser.save()
}
} catch(err){
    //...
}```